app/src/androidTest/java/com/github/se/project/ui/endToEnd/TutorFindLessonEndToEnd.kt

```diff
@@ -0,0 +1,367 @@
+    composeTestRule.onNodeWithTag("firstNameField").assertIsDisplayed()
+
+    // Create Profile Screen
+    composeTestRule.onNodeWithTag("firstNameField").performTextInput("John")
+    composeTestRule.onNodeWithTag("lastNameField").performTextInput("Doe")
+    composeTestRule.onNodeWithTag("phoneNumberField").performTextInput("0213456789")
+    composeTestRule.onNodeWithTag("confirmButton").performScrollTo()
+    composeTestRule.onNodeWithTag("roleButtonTutor").performClick()
+    composeTestRule.onNodeWithTag("sectionDropdown").performClick()
+    composeTestRule.onNodeWithTag("sectionDropdownItem-SC").performClick()
+    composeTestRule.onNodeWithTag("academicLevelDropdown").performClick()
+    composeTestRule.onNodeWithTag("academicLevelDropdownItem-BA3").performClick()
+    composeTestRule.onNodeWithTag("confirmButton").performClick()
+    assertEquals(Role.TUTOR, mockProfileViewModel.currentProfile.value?.role)
+
+    // Create Tutor Profile Screen
+    composeTestRule.onNodeWithTag("welcomeText").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("checkbox_FRENCH").performClick()
+    composeTestRule.onNodeWithTag("checkbox_ENGLISH").performClick()
+    composeTestRule.onNodeWithTag("subjectButton").performClick()
+    composeTestRule.onNodeWithTag("dropdownANALYSIS").performClick()
+    composeTestRule.onNodeWithTag("dropdownPHYSICS").performClick()
+    composeTestRule.onNodeWithTag("subjectButton").performClick()
+    composeTestRule.onNodeWithTag("priceSlider").performGesture { swipeRight() }
+    composeTestRule.onNodeWithTag("confirmButton").performClick()
+
+    // Create Tutor Schedule Screen
+    composeTestRule
+        .onNodeWithTag("welcomeText")
+        .assertTextEquals("John, show us your availabilities")
+    composeTestRule.onNodeWithTag("Slot_0_0").performClick()
+    composeTestRule.onNodeWithTag("Slot_0_3").performClick()
+    composeTestRule.onNodeWithTag("Slot_0_2").performClick()
+    composeTestRule.onNodeWithTag("Slot_0_6").performClick()
+    composeTestRule.onNodeWithTag("FindStudentButton").performClick()
+    mockLessons =
+        listOf(
+            Lesson(
+                id = "1",
+                title = "Maths Tutoring",
+                description = "Fourrier Transform",
+                subject = Subject.ANALYSIS,
+                languages = listOf(Language.ENGLISH),
+                tutorUid = listOf("mockUid"),
+                studentUid = "student123",
+                minPrice = 20.0,
+                maxPrice = 50.0,
+                timeSlot = "16/11/2024T12:00:00",
+                status = LessonStatus.PENDING_TUTOR_CONFIRMATION,
+                latitude = 46.518973490411526,
+                longitude = 6.5685102716088295),
+        )
+    // Call the updatelesson
+    var updatedMockLessonFlow = MutableStateFlow(mockLessons[0])
+    // val updatedMockLessonFlow2 = MutableStateFlow(mockLessons[1])
+    doReturn(updatedMockLessonFlow).`when`(mockLessonViewModel).selectedLesson
```

Comment:
Instead of redefining the way selectedLesson call should respond, one could add the field mockLessonFlow (currently defined in the setup funtion), as a private var of the class that could be simply updated here without having to use the line doreturn()... again.